### PR TITLE
Update mail copy for apply again invitation

### PR DESCRIPTION
**Pull request by stevehook**
## Context

Copy changes to some emails that include a call to action to apply again.

Covers the following emails:
- application_rejected_all_applications_rejected
- application_rejected_by_default_all_applications_rejected
- decline_last_application_choice
- declined_by_default_with_rejections
- declined_by_default_without_rejections
- withdraw_last_application_choice

## Changes proposed in this pull request

- [x] Copy updates

### application_rejected_all_applications_rejected
image

### application_rejected_by_default_all_applications_rejected
image

### decline_last_application_choice
image

### declined_by_default_with_rejections
image

### declined_by_default_without_rejections
image

### withdraw_last_application_choice
image

- [x] Refactor to DRY templates

## Guidance to review

- Do the screenshots (or the review app) match the requirements?
- Does the refactoring make sense?

## Link to Trello card

https://trello.com/c/7hMv8M5e/3032-changes-to-email-for-applying-again-in-same-cycle

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training/blob/master/docs/environment-variables.md#azure-hosting-devops-pipeline)

### CODE REVIEW

**Review by Tomcowan1 — APPROVED** [review]

File: app/views/candidate_mailer/application_rejected_all_applications_rejected.text.erb [hunk 1]
```diff
@@ -20,6 +20,9 @@ Contact <%= @course.provider.name %> directly if you have any questions about th
   You’ve now had decisions about all the courses you applied for. You did not get any offers, so you can apply again.
 <% end %>
 
-Your last application has been saved. You can make changes before you submit your new application.
+If now’s the right time for you, you can apply for teacher training again this year. 
 
+Your last application has been saved. You can make changes and choose a different course before you submit a new application.
+
+Sign in to apply again:
```
Comment (raamSoftwire) [inline app/views/candidate_mailer/application_rejected_all_applications_rejected.text.erb:27]: For the sake of consistency, should there be an empty line here before the magic link? 

(I'm not sure if that would get rendered in the email, so not sure if it matters.)

Comment (raamSoftwire) [reply app/views/candidate_mailer/application_rejected_all_applications_rejected.text.erb:27]: Based on the screenshot above, I think the extra line does get rendered.

File: app/views/candidate_mailer/application_rejected_all_applications_rejected.text.erb [hunk 2]
```diff
@@ -20,6 +20,9 @@ Contact <%= @course.provider.name %> directly if you have any questions about th
   You’ve now had decisions about all the courses you applied for. You did not get any offers, so you can apply again.
 <% end %>
 
-Your last application has been saved. You can make changes before you submit your new application.
+If now’s the right time for you, you can apply for teacher training again this year.
```
Comment (raamSoftwire) [inline app/views/candidate_mailer/application_rejected_all_applications_rejected.text.erb:23]: Other than the title this is all shared with the `_interested` partial. Is it worth pulling it out further? (The answer might be no, just wanted to ask the question).

File: app/views/candidate_mailer/application_rejected_all_applications_rejected.text.erb (hunk 1 above)
Comment (stevehook) [reply app/views/candidate_mailer/application_rejected_all_applications_rejected.text.erb:27]: Good spot. Thanks!

**Review by raamSoftwire — APPROVED** [review]
Cheers Steve, this looks good

File: app/views/candidate_mailer/application_rejected_all_applications_rejected.text.erb (hunk 2 above)
Comment (stevehook) [reply app/views/candidate_mailer/application_rejected_all_applications_rejected.text.erb]: Fair point. Have refactored a bit DRY things a bit.

**Pull request merged**